### PR TITLE
jackson support for score (based on Christopher's PR)

### PR DESCRIPTION
**Pull request by ge0ffrey**

### CODE REVIEW

File: optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/PolymorphicScoreJacksonJsonSerializer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
```
Comment (yurloc) [inline optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/PolymorphicScoreJacksonJsonSerializer.java]: Should be 2018 unless the file has already existed an was just modified & moved.

File: optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/OptaPlannerJacksonModule.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+     * @return never null, register it with {@link ObjectMapper#registerModule(Module)}.
+     */
+    public static Module createModule() {
+        SimpleModule module = new SimpleModule("OptaPlanner");
+
+        // For non-subtype Score fields/properties, we also need to record the score type
+        module.addSerializer(Score.class, new PolymorphicScoreJacksonJsonSerializer());
+        module.addDeserializer(Score.class, new PolymorphicScoreJacksonJsonDeserializer());
+
+        module.addSerializer(SimpleScore.class, new SimpleScoreJacksonJsonSerializer());
```
Comment (Christopher-Chianelli) [inline optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/OptaPlannerJacksonModule.java]: Might be able to auto generate these with generators: http://hannesdorfmann.com/annotation-processing/annotationprocessing101

File: optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/PolymorphicScoreJacksonJsonDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ */
+public class PolymorphicScoreJacksonJsonDeserializer extends JsonDeserializer<Score> {
+
+    @Override
+    public Score deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        parser.nextToken();
+        String scoreClassSimpleName = parser.getCurrentName();
+        parser.nextToken();
+        String scoreString = parser.getValueAsString();
+        if (scoreClassSimpleName.equals(SimpleScore.class.getSimpleName())) {
```
Comment (Christopher-Chianelli) [inline optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/PolymorphicScoreJacksonJsonDeserializer.java]: Might be able to generate these with generators, see above

File: optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/buildin/bendable/BendableScoreJacksonJsonSerializer.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+
+@SuppressWarnings("checkstyle:javadocstyle")
+/**
+ * {@inheritDoc}
+ */
+public class BendableScoreJacksonJsonSerializer extends ScoreJacksonJsonSerializer<BendableScore> {
```
Comment (Christopher-Chianelli) [inline optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/buildin/bendable/BendableScoreJacksonJsonSerializer.java]: Might be able to generate all of these classes with generators, see above

File: optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/OptaPlannerJacksonModule.java (hunk 2 above)
Comment (ge0ffrey) [reply optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/OptaPlannerJacksonModule.java]: Those lines? Let's keep it simple, so our IDE's "find usages" work, our code is easy to debug (there is source code) and our dependencies/build are simple.
(Those kind of smart tricks tend to leak into the user's experience and ruin that.)

**Pull request merged**